source/Irrlicht/CMakeLists.txt

```diff
@@ -406,6 +374,23 @@ add_library(IRRVIDEOOBJ OBJECT
 	${IRRIMAGEOBJ}
 )
 
+target_link_libraries(IRRVIDEOOBJ
+	PUBLIC
+		# TODO Create imported targets if they do not already exist,
+		# and only link the correct one for the enabled backend.
+		"$<$<BOOL:${OPENGL_DIRECT_LINK}>:${OPENGL_LIBRARIES}>"
+		"$<$<BOOL:${OPENGLES_DIRECT_LINK}>:${OPENGLES_LIBRARY}>"
+		"$<$<BOOL:${OPENGLES2_DIRECT_LINK}>:${OPENGLES2_LIBRARIES}>"
+		"${EGL_LIBRARY}"
+		JPEG::JPEG
+	PRIVATE
```

Comment:
by which logic is public/private decided here?

Reply:
I usually base it on whether the dependency is included from a public header. I will double check all of these before taking it out of draft.